src/sqlfluff/utils/reflow/README.md

```diff
@@ -0,0 +1,219 @@
+
+In general there are two phases to a reflow related rule:
+1. _Linting_ - working out whether there's an issue.
+2. _Fixing_ - resolving the issue.
+
+The intent is that _Fixing_ should always be in the reflow code.
+_Linting_ may or may not be in the reflow code and could also be
+encoded within the rule itself.
+
+### How to apply in a rule context
+
+When applying reflow in a rule context it makes sense to consider
+how widespread the impact we want to have is:
+- A rule designed around _Spacing_ may act in relative isolation,
+  but may also optionally need to resolve excessive _Line Length_.
+- A rule designed around _Indentation_ may act simply on it's own
+  but may also optionally need to resolve excessive _Line Length_.
+- A rule designed around the placement or presence of _Line Breaks_
+  will probably have to consider _Spacing_ (or at least decide
+  what spacing to impose at least for the moved elements) and if
+  introducing additional _Line Breaks_ it may need to account
+  for _Indentation_.
+- Resolving _Line Length_ will involve introducing additional
+  _Line Breaks_ (assuming that _Spacing_ has already been covered).
+
+As a rule of thumb:
+- If it's a rule about _Spacing_ it should only care about that
+  in full knowledge that it may make lines too long. Unless the
+  user has _also_ enabled long line checking, this is fine, and
+  if they have enabled it - that rule will pick things up itself.
+  - This means we'll likely need a `.respace()` method.
+- If it's a rule about _Indentation_ we should also not check
+  line length (for the same reasons).
+  - This means we'll likely need a `.reindent()` method.
+- If it's a rule about _Line Breaks_ in relation to other segments
+  (e.g. around commas or operators), we should aim to keep the
+  existing _Indentation_ and not reset that or consider any
+  _Line Length_ issues. We **should** consider _Spacing_, but
+  only around the elements moved (e.g. if we're shifting a trailing
+  comma to be a leading one, then we should also correct it's spacing
+  at the same time).
+  - This means in practice we should move the _Segment_ and not
+    move the _Line Break_. Once moved, we should use the `.respace()`
+    method to resolve the spacing.
+- If it's a rule about _Line Length_ we will also need to consider
+  _Indentation_ and _Line Breaks_.
+  - This should use a more fully featured `.reflow()` method which
+    can use `.reindent()` and others under the hood, but limited
+    only to reflowing the offending line itself (TBC whether in
+    in practice this needs to be broader).
+
+### What about non-reflow rules?
+
+Some rules add, remove or rearrange segments and need to do so in
+a way which is reflow aware.
+- In general, rules which only _remove_ segments as part of a fix
+  should only readdress the _Spacing_ left in the gap. We should
+  not attempt to reflow and attempt to remove any _Line Breaks_
+  which could now be possible due to the remaining shorter lines.
```

Comment:
Do you mean **never remove line breaks", or that it would happen elsewhere/later?

Reply:
Correct. This was lazy use of the word `not`. Will correct it.